
<br>

# Workshop Project 

This project will be completed in the two days of the workshop (August 28-29). We will assign attendess into groups to encourage discussions on approaching each quesstion. Everyone must do their own analysis and submit the final R code.

<br>

## Project Overview
We will be working on a small subset of the TCGA Colon cancer dataset. The Cancer Genome Atlas (TCGA), is a collaboration between different research bodies to identify key genomic changes in various cancer types. These datasets are standards for hypothesis testing and validating findings.

We will be using a subet of the colon RNASeq dataset and clinical data. Following are the questions that we would like you to answer. There is no one way of answering these questions so feel free to use any of the many ways you have learnt to answer each question!

<br>

## Download Data
Please download the Project data folder from:

https://www.dropbox.com/sh/0iqhknngbugiqcc/AACKnm768SXDeZI708xmn2Qla?dl=0

.You should have two files: TCGA_clinical.csv and TCGA_expression_sample.csv

## Questions

1. Read in the two datafiles into R and using one of the built-in R functions look at the file structure i.e. figure out what is in the files.        
      + Find out how many samples are there in each file. 
      + Do all the sample names in the clinical and expression file match?

<br>

2. Find out if there is any unnecessary information in the clinical file. Hint: Is there a column which does not provide any information? If so, please remove it from the file.

<br>

3. For the gene expression data, find out where are the gene ids and assign gene ids as row names. Also remove the gene_id column after assigning the ids to rownames. 

<br>

4. Create a boxplot showing the distribution of the log of read counts for each sample. Make the x-axis lables verticle to the axis. A possible example is as follows:

```{r,echo=FALSE,message=FALSE,warning=FALSE}
library(tidyverse)
clinical = read_csv("ProjectData/TCGA_clinical.csv") 
genes = read_csv("ProjectData/TCGA_expression_sample.csv") 

rownames(genes) = genes$gene
genes = genes[, which(colnames(genes) != "gene")]

# Explore
#summary(clinical)
#apply(clinical, 2, table)

# get rid of grade column - it's empty
clinical = clinical[, which(colnames(clinical) != "grade")]


## verctical labels
genes %>% 
  rownames_to_column("gene") %>% 
  gather(sample, count, -gene) %>% 
  ggplot(aes(x = sample, y = log2(count + 1))) +
  geom_boxplot() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))

```

<br>

5. Point out a few characteristics of these boxplots.

<br>

6. As this expression data is count data, remove all the zeros from the expression matrix. Now, remake the boxplot from question 4.

<br>

7. Lets make things more exciting! Next we will make a Principle Component Analysis (PCA) plot to see the relationship between samples and color the samples by type.
	* For those who are not familiar with PCA, it is a dimensionality reduction technique that is widely used in data analysis. Reducing the dimensionality of a dataset can be useful in different ways. For example, our ability to visualize data is limited to 2 or 3 dimensions. Lower dimension can sometimes significantly reduce the computational time of some numerical algorithms. Besides, many statistical models suffer from high correlation between covariates, and PCA can be used to produce linear combinations of the covariates that are uncorrelated between each other.
	* There are a few ways in R to perform a PCA analysis. One function which is found in basic R stats package is called <b> prcomp </b>. Take a look at the help for this function and then answer the following questions.
	* As we want to make the PCA plot for samples,samples should be rows in your input matrix to prcomp. Check if that is the case in your expression matrix. If not modify it. Hint: look at how to transpose the matrix.
	* Using the help for prcomp function see what scale and center attributes do? Now look online to find out if your data needs to be centered or scaled.
	* Use the prcomp function on your expression matrix (with appropriate attributes) and save the output into an object. Next, explore this object to see what information is stored in this. Try to find out which variable in this object holds the information about principle components(PC) and how many PCs do you have in total
	* Once you know which variable holds the information about PCs, save the first two components into a separate object and use them to make the pCA plot in the next step.
	* Make the PCA plot for the first two components (as shown in the plot below) for samples and color them using type column in clinical file.
	* Are the samples separated by type? Are there any subgroups within each subtype?
	* Put sample labels on your PCA plot.

<br>

Example of one of the possible PCA plots is as follows:

```{r,echo=FALSE,message=FALSE,warning=FALSE}
genes_filt = genes[rowSums(genes) > 0, ]

pca = prcomp(t(log2(genes_filt + 1)), scale. = TRUE)
pca_data = as.data.frame(pca$x)

pca_data %>% 
  rownames_to_column("sample") %>% 
  left_join(clinical) %>% 
  ggplot(aes(x = PC1, y = PC2,label=sample)) +
  geom_point(aes(colour = type)) +
  geom_label(aes(fill = factor(type)), colour = "white", fontface = "bold")
```

<br>

8. Usually for gene expression data, we want to see how the gene expression levels have changed in tumor samples compared to normal samples.Let us explore that by calculating log fold change for all genes in these samples.
    + Find out the log2fold changes from normal to Tumor samples. Hint: You will first have to calculate the mean for each sample type and then calculate log2FC log2(mean(type1)/mean(type2)+1). Hint: A useful function here can be rowMeans.
    * Find out the gene with the highest log2FC.
    * Make a plot for this gene  with normal and Tumor sample expression. 

<br>

9. You must have noticed by now that normal samples have longer names than the tumor samples. Is there a reason for that? 
    + Check if we can remove the extra information in the normal sample names i.e. check if the longer names make the samples name unique or is it redundant information and we can get rid of it?
    


